contracts/extensions/ERC721AQueries.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+// Creator: Chiru Labs
+
+pragma solidity ^0.8.4;
+
+import '../ERC721A.sol';
+
+/**
+ * @title ERC721A Queries
+ * @dev ERC721A subclass with convenience query functions.
+ */
+abstract contract ERC721AQueries is ERC721A {
+    /**
+     * @dev Returns the `TokenOwnership` struct at `tokenId` without reverting.
+     *
+     * If the `tokenId` is out of bounds:
+     *   - `addr` = `address(0)`
+     *   - `startTimestamp` = `0`
+     *   - `burned` = `false`
+     *
+     * If the `tokenId` is burned:
+     *   - `addr` = `<Address of owner before token was burned>`
+     *   - `startTimestamp` = `<Timestamp when token was burned>`
+     *   - `burned = `true`
+     *
+     * Otherwise:
+     *   - `addr` = `<Address of owner>`
+     *   - `startTimestamp` = `<Timestamp of start of ownership>`
+     *   - `burned = `false`
+     */
+    function rawOwnershipOf(uint256 tokenId) public view returns (TokenOwnership memory) {
+        if (_exists(tokenId)) {
```

Comment:
Wouldn't it be better to fetch `_ownerships[tokenId]` first, store it in memory and then reading from it? Right now `_exists` does an SLOAD and then line 36 will result in a second SLOAD.

Reply:
Ok, good suggestion.